library/azure_rm_trafficmanagerendpoint.py

```diff
@@ -0,0 +1,358 @@
+        name=endpoint.name,
+        type=endpoint.type,
+        target_resource_id=endpoint.target_resource_id,
+        target=endpoint.target,
+        status=endpoint.endpoint_status,
+        weight=endpoint.weight,
+        priority=endpoint.priority,
+        location=endpoint.endpoint_location,
+        monitor_status=endpoint.endpoint_monitor_status,
+        min_child_endpoints=endpoint.min_child_endpoints,
+        geo_mapping=endpoint.geo_mapping
+    )
+    # if endpoint.geo_mapping:
+
+    #     for geo in endpoint.geo_mapping:
+    #         result['geo_mapping'].append(geo)
+    )
+
+
+class Actions:
+    NoAction, CreateOrUpdate, Delete = range(3)
+
+class AzureRMTrafficManagerEndpoint(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            profile_name=dict(
+                type='str',
+                required=True
+            ),
+            type=dict(
+                type='str',
+                choices=['azureEndpoints', 'externalEndpoints', 'nestedEndpoints'],
+                required=True
+            ),
+            target_resource_id=dict(type='str'),
+            enable=dict(type='bool', default=True),
+            weight=dict(type='int'),
+            priority=dict(type='int'),
+            location=dict(type='str'),
+            minitor_status=dict(type='str'),
+            min_child_endpoints=dict(type='int'),
+            geo_mapping=dict(type='list', elements='str'),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+
+        self.profile_name = None
+        self.type = None
+        self.target_resource_id = None
+        self.enable = None
+        self.weight = None
+        self.priority = None
+        self.location = None
+        self.minitor_status = None
+        self.min_child_endpoints = None
+        self.geo_mapping = None
+        self.endpoint_status = 'Enabled'
+
+        self.action = Actions.NoAction
+
+        self.results = dict(
+            changed=False
+        )
+
+        super(AzureRMTrafficManagerEndpoint, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             mutually_exclusive=mutually_exclusive,
+                                             supports_check_mode=True,
+                                             supports_tags=False)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+
+        to_be_updated = False
+
+        if not self.location:
+            self.location = 'global'
```

Comment:
traffic manager endpoint can have location like 'WestUS'. I tested use 'global' as endpoint_location and got error msg that global is invalid.